db/migrate/20171103121704_add_entry_date_to_client.rb

```diff
@@ -1,6 +1,9 @@
 class AddEntryDateToClient < ActiveRecord::Migration[5.1]
   def change
-    remove_column :clients, :entry_year, :date
     add_column :clients, :entry_date, :string
+    Client.find_each do |client|
+      client.update(entry_date: client.entry_year)
+    end
```

Comment:
Note that this code will run both when running the migration and rolling it back, you can avoid that by defining `up` / `down` methods separately. 

Also I'm not sure if it will recognize the new `entry_date` column immediately, it used to be that you have to call `Client.reset_column_information` in the migration after adding new columns but I'm not sure if that's still the case in Rails 5. If the migration works locally it should be fine :)

Reply:
It works locally. The thing is that in case of a rollback, how can the data be saved since the old column is `:date` and the new is `:string` and string cannot be stored in date?